### PR TITLE
Issue 222

### LINKED ISSUE #222: Optimize full-text index queries
This is currently the largest bottleneck in the probabilistic workflow, and also affects the baseline.

- [x] `data_gathering.tokens_fulltext_search` should take as input a batch of tokens and run a bulk query;
- [x] #205 , which should also apply the above task to `data_gathering.name_fulltext_search`;
- [x] #207

### PR DESCRIPTION
**Pull request by MaxFrax**
Closes #222

### CODE REVIEW

File: soweego/linker/blocking.py [hunk 1]
```diff
@@ -75,22 +70,23 @@ def full_text_query_block(goal: str, catalog: str, wikidata_series: pd.Series, c
...
-    for result in tokens_fulltext_search(target_entity, boolean_mode, terms):
-        tids.add(result.catalog_id)
-        LOGGER.debug('%s result: %s',
-                     target_entity.__name__, result)
 
-    if not tids:
-        LOGGER.info('No target candidates for source query: %s', terms)
-        return nan
+def fulltext_search(qid_terms_target: Tuple[str, list, constants.DB_ENTITY]) -> Iterable[Tuple[str, str]]:
+    qid = qid_terms_target[0]
```
Comment (marfox) [inline soweego/linker/blocking.py]:
```suggestion
    qid, terms, target_entity = qid_terms_target
```
Just a little bit more Pythonic 😄

File: soweego/linker/blocking.py [hunk 2]
```diff
@@ -75,22 +70,23 @@ def full_text_query_block(goal: str, catalog: str, wikidata_series: pd.Series, c
...
+    terms = qid_terms_target[1]
```
Comment (marfox) [inline soweego/linker/blocking.py]:
```suggestion
```
(multiline suggestions are currently not supported)

File: soweego/linker/blocking.py [hunk 3]
```diff
@@ -75,22 +70,23 @@ def full_text_query_block(goal: str, catalog: str, wikidata_series: pd.Series, c
...
+    target_entity = qid_terms_target[2]
```
Comment (marfox) [inline soweego/linker/blocking.py]:
```suggestion
```
(multiline suggestions are currently not supported)

File: soweego/linker/blocking.py [hunk 4]
```diff
@@ -75,22 +70,23 @@ def full_text_query_block(goal: str, catalog: str, wikidata_series: pd.Series, c
...
+    ids = list(map(lambda entity: entity.catalog_id, tokens_fulltext_search(target_entity, False, terms, None, 5)))
```
Comment (marfox) [inline soweego/linker/blocking.py]:
```suggestion
    tids = list(map(lambda entity: entity.catalog_id, tokens_fulltext_search(target_entity, False, terms, None, 5)))
```

File: soweego/linker/blocking.py [hunk 5]
```diff
@@ -75,22 +70,23 @@ def full_text_query_block(goal: str, catalog: str, wikidata_series: pd.Series, c
...
+    return [(qid, id) for id in ids]
```
Comment (marfox) [inline soweego/linker/blocking.py]:
```suggestion
    return [(qid, tid) for tid in tids]
```
Just avoiding a Python keyword override (`id`)

File: soweego/linker/blocking.py [hunk 6]
```diff
@@ -23,7 +26,7 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def train_test_block(wikidata_df: pd.DataFrame, target_df: pd.DataFrame) -> pd.MultiIndex:
+def train__block(wikidata_df: pd.DataFrame, target_df: pd.DataFrame) -> pd.MultiIndex:
```
[the remark this answers is not in the document]
Comment (MaxFrax) [reply soweego/linker/blocking.py]: Probably something has gone wrong while finding/replacing the term test. Sorry! Will fix soon

File: soweego/linker/blocking.py (hunk 5 above)
Comment (MaxFrax) [reply soweego/linker/blocking.py]: Actually, I removed it while refactoring by mistake. Probably, it's safer to put it back

**Comment by marfox** [comment]
> could you please explain why you removed the null values dropping (i.e., the `dropna`s in lines 54 and 56)?

>> Actually, I removed it while refactoring by mistake. Even though I didn't have issues while testing, it is probably safer/harmless to put it back.

@MaxFrax : no worries, got it.

**Pull request merged**